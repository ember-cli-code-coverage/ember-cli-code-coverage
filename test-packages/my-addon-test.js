'use strict';

const fs = require('fs-extra');
const util = require('util');
const rimraf = util.promisify(require('rimraf'));
const { dir, file } = require('chai-files');
const path = require('path');
const execa = require('execa');

const BASE_PATH = path.join(__dirname, 'my-addon');

describe('addon coverage generation', function () {
  jest.setTimeout(10000000);

  beforeEach(async function () {
    await rimraf(`${BASE_PATH}/coverage*`);
    await execa('git', ['clean', '-f', 'my-addon'], { cwd: __dirname });
    await execa('git', ['restore', 'my-addon'], { cwd: __dirname });
  });

  afterEach(async function () {
    await rimraf(`${BASE_PATH}/coverage*`);
    await execa('git', ['clean', '-f', 'my-addon'], { cwd: __dirname });
    await execa('git', ['restore', 'my-addon'], { cwd: __dirname });
  });

  it('runs coverage on templates when option is set', async function() {
    dir(`${BASE_PATH}/coverage`).assertDoesNotExist();
    fs.copySync(`${BASE_PATH}/config/-coverage-template.js`, `${BASE_PATH}/config/coverage.js`);

    let env = { COVERAGE: 'true' };
    await execa('yarn', ['test'], { cwd: BASE_PATH, env });

    let summary = fs.readJSONSync(`${BASE_PATH}/coverage/coverage-summary.json`);
    expect(summary).toMatchSnapshot();
  });
});
