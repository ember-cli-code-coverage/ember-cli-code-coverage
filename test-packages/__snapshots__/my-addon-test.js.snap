// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app coverage generation runs coverage on templates when option is set 1`] = `
Object {
  "addon-test-support/uncovered-test-support.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 2,
    },
    "functions": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 2,
    },
    "lines": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 4,
    },
    "statements": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 4,
    },
  },
  "addon/components/addon-component.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "lines": Object {
      "covered": 2,
      "pct": 100,
      "skipped": 0,
      "total": 2,
    },
    "statements": Object {
      "covered": 2,
      "pct": 100,
      "skipped": 0,
      "total": 2,
    },
  },
  "addon/utils/my-covered-util.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "statements": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
  },
  "addon/utils/my-uncovered-util.js": Object {
    "branches": Object {
      "covered": 0,
      "pct": 100,
      "skipped": 0,
      "total": 0,
    },
    "functions": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "lines": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
    "statements": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 1,
    },
  },
  "total": Object {
    "branches": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 2,
    },
    "functions": Object {
      "covered": 0,
      "pct": 0,
      "skipped": 0,
      "total": 4,
    },
    "lines": Object {
      "covered": 2,
      "pct": 25,
      "skipped": 0,
      "total": 8,
    },
    "statements": Object {
      "covered": 2,
      "pct": 25,
      "skipped": 0,
      "total": 8,
    },
  },
}
`;
